src/main/scala/mesosphere/marathon/core/launchqueue/impl/LaunchQueueActor.scala

```diff
@@ -0,0 +1,186 @@
+package mesosphere.marathon.core.launchqueue.impl
+
+import akka.actor.SupervisorStrategy.Stop
+import akka.actor.{
+  PoisonPill,
+  Terminated,
+  Actor,
+  ActorLogging,
+  ActorRef,
+  OneForOneStrategy,
+  Props,
+  SupervisorStrategy
+}
+import akka.event.LoggingReceive
+import akka.pattern.{ ask, pipe }
+import akka.util.Timeout
+import mesosphere.marathon.core.launchqueue.LaunchQueue
+import mesosphere.marathon.state.{ AppDefinition, PathId }
+import LaunchQueue.QueuedTaskCount
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+
+private[launchqueue] object LaunchQueueActor {
+  def props(appActorProps: (AppDefinition, Int) => Props): Props = {
+    Props(new LaunchQueueActor(appActorProps))
+  }
+
+  case class FullCount(appId: PathId)
+}
+
+/**
+  * An actor-based implementation of the [[LaunchQueue]] interface.
+  *
+  * The methods of that interface are translated to messages in the [[ActorLaunchQueue]] implementation.
+  */
+private[impl] class LaunchQueueActor(appActorProps: (AppDefinition, Int) => Props) extends Actor with ActorLogging {
+  import ActorLaunchQueue._
+
+  /** Currently active actors by pathId. */
+  var launchers = Map.empty[PathId, ActorRef]
+  /** Maps actorRefs to the PathId they handle. */
+  var launcherRefs = Map.empty[ActorRef, PathId]
+
+  /** Serial ID to ensure unique names for children actors. */
+  var childSerial = 0
+
+  // See [[receiveHandlePurging]]
+  /** A message with a sender for later processing. */
+  case class DeferredMessage(sender: ActorRef, message: Any)
+  /** PathIds for which the actors have been currently suspended because we wait for their termination. */
+  var suspendedLauncherPathIds = Set.empty[PathId]
+  /** ActorRefs of the actors have been currently suspended because we wait for their termination. */
+  var suspendedLaunchersMessages = Map.empty[ActorRef, Vector[DeferredMessage]].withDefault(_ => Vector.empty)
+
+  /** The timeout for asking any children of this actor. */
+  implicit val askTimeout: Timeout = 3.seconds
+
+  override def receive: Receive = LoggingReceive {
+    Seq(
+      receiveHandlePurging,
+      receiveMessagesToSuspendedActor,
+      receiveHandleNormalCommands
+    ).reduce(_.orElse[Any, Unit](_))
+  }
+
+  /**
+    * Handles purging of an actor.
+    *
+    * Terminating an actor with a PoisonPill is not instantaneous. It will still process
```

Comment:
Do I understand that correctly:
`DELETE /v2/app/foo`
`POST /v2/app/foo`
If delete takes some time, the same actor will handle create?